doc/design/vic-machine/update.md

```diff
@@ -0,0 +1,175 @@
+
+## Update In Progress Status
+Same to vic-machine upgrade, concurrent reconfigure requests for same VCH is not allowed. vic-machine should be able to detect if one VCH is running reconfigure/upgrade. If true, return useful error message. And also should not leave that reconfigure flag in there after reconfigure is stopped, or even interrupted.
+
+Issue https://github.com/vmware/vic/issues/4069 is tracking this problem.
+
+## Backward compatible
+It's possible that after vic engine released new version, user is trying to reconfigure older version's VCH with newer vic-machine.
+
+There are few options:
+- Run reconfigure using same version's vic-machine (this option should technically always work)
+
+  For any customer that does not want to change vic-machine version, they need to run reconfigure using same version's vic-machine.
+
+- Combine upgrade process into reconfigure
+
+  Upgrade is actually one special kind of reconfigure, technically it's easy to have upgrade run together with reconfigure. User might want to reconfigure VCH, and also want to upgrade it after or before reconfigure. In that case, a combined reconfigure and upgrade command can provide a better user experience.
+
+ Here is the combined command ```vic-machine reconfigure --upgrade <all reconfigure and upgrade options supported here>```
+
+- Support backward compatibility in vic-machine
+
+  This will need to add backward compatible code in vic-machine. Even initially there is only few changes, that will expand massively over time, which is not consistent with our data migration design, so this is not our option.
+
+# VCH reconfigure Rollback (P0.5)
+Similar to VCH upgrade workflow, before reconfigure VCH, snapshot should be took to keep existing status. If the reconfigure failed eventually, VCH should be reverted to old status automatically.
+
+The Reconfigure Rollback is talking about after a successful reconfigure, vic-machine is still able to rollback VCH's old configuration. The command is like vic-machine reconfigure --rollback
+
+This feature is nice to have, but as vic-machine upgrade already have this capability, mark it as P0.5.
+
+# Manifest File
+vic-machine create already has more than 50 options, and the number will increase over time. With this reconfigure feature added, most of those options need to be updated as well. To improve user experience, in my opinion, we need Manifest file to simplify vic-machine command generation.
+
+[vic-machine design](vic-machine.md#vch-manifest) mentioned manifest idea, here we'll start from a simple implementation. The manifest file format will be same to the yml file format, mentioned in [configuration inspection](#configuration-inspection)
+
+- vic-machine inspect --conf --manifest filename will dump all configurations in yml file format to the manifest file.
```

Comment:
hmm.. should be fine. then all options for `vic-machine inspect` will be as following:
- `vic-machine inspect` will show the VCH endpoint information, which is current behavior
- `vic-machine inspect --cmd` will dump command options format configuration
- `vic-machine inspect --conf` will dump yml file format configuration
- `vic-machine inspect --manifest` will dump yml format configuration into manifest file